include/revng/Model/QualifiedType.h

```diff
@@ -62,6 +63,17 @@ class model::QualifiedType : public model::generated::QualifiedType {
   bool verify(bool Assert) const debug_function;
   RecursiveCoroutine<bool> verify(VerifyHelper &VH) const;
   void dump() const debug_function;
+
+  bool operator==(const QualifiedType &) const = default;
```

Comment:
Yes, in this case `operator==` is faster, because it does not lexicographically compare the whole things to see if one is smaller than the other. If they are different it returns false and bails out early.  In order to tell the compiler to do this fast codegen of `operator==` you have either to have a defaulted `operator<=>` or to default `operator==` explicitly. The problem here is that we have a non-defaulted `operator<=>`, and we cannot default it because `Qualifiers` is a `std::vector` and `std::vector` does not have `operator<=>` in the version of the STL we're using.